test/gtest/ucp/ucp_test.cc

```diff
@@ -157,6 +158,60 @@ void ucp_test::flush_worker(const entity &e, int worker_index)
     wait(request, worker_index);
 }
 
+void ucp_test::flush(ucs::ptr_vector<ucp_test_base::entity> *entities)
```

Comment:
can be simplified by `ucp_test_base::entity::flush_worker_nb` + `test_ucp_wireup::waitall`. Many branches are redundant.

Reply:
good catch! Thanks!

waitall was moved to UCP common code (`test_ucp_wireup::waitall` -> `ucp_test::waitall`) as it does common things.